### PR TITLE
Early return `select()`, `filter()`, and `arrange()`

### PR DESCRIPTION
**Pull request by mgirlich**
Return early if the operation doesn't change the data.

### CODE REVIEW

File: R/verb-select.R [hunk 1]
```diff
@@ -101,6 +101,10 @@ simulate_vars <- function(x) {
 # op_select ---------------------------------------------------------------
 
 op_select <- function(x, vars) {
+  if (can_select_return_early(x, vars)) {
```
Comment (hadley) [inline R/verb-select.R:104]: Again, I think I'd rather give this a more descriptive name, like `selects_same_variables()` maybe?

File: tests/testthat/test-verb-arrange.R [hunk 2]
```diff
@@ -76,6 +76,12 @@ test_that("can combine arrange with dual table verbs", {
   })
 })
 
+test_that("only add step if necessary", {
+  lf <- lazy_frame(x = 1:3, y = 1:3)
+  expect_equal(lf %>% arrange(), lf)
+  expect_equal(lf %>% filter(!!!list()), lf)
```
Comment (hadley) [inline tests/testthat/test-verb-arrange.R:82]: Why test `filter` here?

File: tests/testthat/test-verb-filter.R [hunk 3]
```diff
@@ -29,6 +29,12 @@ test_that("each argument gets implicit parens", {
   expect_equal_tbl(mf1, mf2)
 })
 
+test_that("only add step if necessary", {
+  lf <- lazy_frame(x = 1:3, y = 1:3)
+  expect_equal(lf %>% filter(), lf)
+  expect_equal(lf %>% filter(!!!list()), lf)
```
Comment (hadley) [inline tests/testthat/test-verb-filter.R:35]: Why can't this be `lf %>% filter()`?

**Review by hadley — APPROVED** [review]
I've approved this so you're free to merge, but do take a look at the comment and make changes if needed (but I don't need to review again, just use your best judgement)

File: tests/testthat/test-verb-filter.R (hunk 3 above)
Comment (mgirlich) [reply tests/testthat/test-verb-filter.R:35]: Hmm, there was a case where `some_verb()` (I don't know which verb it was) and `some_verb(!!!list())` was different. But I don't see where this should have been the case so I remove these tests.

**Pull request merged**